src/main/java/seedu/address/logic/commands/SetGoalCommand.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Goal;
+import seedu.address.model.person.Person;
+
+public class SetGoalCommand extends Command {
+
+    public static final String COMMAND_WORD = "set-goal";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Set goals for meeting the person identified by the index.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "FREQUENCY (w[eek[ly]] | m[onth[ly]] | y[ear[ly]] | n[one])\n"
+            + "Example: " + COMMAND_WORD + " 1 f/w";
+
+    public static final String MESSAGE_ADD_GOAL_SUCCESS = "Set %1$s goal for %2$s";
+
+    private final Index index;
+    private final Goal.Frequency frequency;
```

Comment:
Can just do static import, then it will look like `private final Frequency frequency`